diff --git a/dist/pnpm/doPnpmBumpVersion.js b/dist/pnpm/doPnpmBumpVersion.js
index 2470346f66209b9b41ccafdab3e4c8a1d9201270..83ce095b3e5380f86d354c1e53f8ecbb33a1b325 100644
--- a/dist/pnpm/doPnpmBumpVersion.js
+++ b/dist/pnpm/doPnpmBumpVersion.js
@@ -7,5 +7,5 @@ import { runOnChanged } from "./utils/runOnChanged.js";
  * @param excludePackages filter out the specified packages/directory from the output
  */
 export default async function (newVersion, since, forcePackages = [], excludePackages = []) {
-    await runOnChanged(`pnpm version ${newVersion} --no-git-tag-version`, since, forcePackages, excludePackages);
+    await runOnChanged(`pnpm version ${newVersion} --no-git-tag-version --no-git-checks`, since, forcePackages, excludePackages);
 }
diff --git a/dist/pnpm/doPnpmPublishVersions.js b/dist/pnpm/doPnpmPublishVersions.js
index 107a330e8969397754d20be00bd34bc432240be0..3ec157e1d808115432a6571ea9e7bd69ef0d1bd8 100644
--- a/dist/pnpm/doPnpmPublishVersions.js
+++ b/dist/pnpm/doPnpmPublishVersions.js
@@ -13,6 +13,7 @@ import pnpmLogger from "./utils/pnpmLogger.js";
  */
 export default async function (since, tag, branch, forcePackages = [], excludePackages = []) {
     const filters = since ? `...[${since}]` : undefined;
-    const pnpmArgs = ["--recursive"].concat(getOptionalFlagArgument("--filter", filters), ...getIncludeFilters(forcePackages), ...getExclusionFilters(excludePackages), "publish", getOptionalFlagArgument("--tag", tag), getOptionalFlagArgument("--publish-branch", branch));
+    const pnpmArgs = ["--recursive"].concat(getOptionalFlagArgument("--filter", filters), ...getIncludeFilters(forcePackages), ...getExclusionFilters(excludePackages), "exec", "npm", // https://github.com/pnpm/pnpm/issues/11098
+    "publish", getOptionalFlagArgument("--tag", tag), "--access", "public", "--provenance");
     await spawn("pnpm", pnpmArgs, pnpmLogger);
 }
diff --git a/src/pnpm/doPnpmBumpVersion.ts b/src/pnpm/doPnpmBumpVersion.ts
index 7865dbff9de6bf2f2a5219e320249ff053a6233e..e6e5f7d9c30351c47d175e1f4c7b2f3cb8ed53e0 100644
--- a/src/pnpm/doPnpmBumpVersion.ts
+++ b/src/pnpm/doPnpmBumpVersion.ts
@@ -14,7 +14,7 @@ export default async function (
   excludePackages: string[] = [],
 ) {
   await runOnChanged(
-    `pnpm version ${newVersion} --no-git-tag-version`,
+    `pnpm version ${newVersion} --no-git-tag-version --no-git-checks`,
     since,
     forcePackages,
     excludePackages,
diff --git a/src/pnpm/doPnpmPublishVersions.ts b/src/pnpm/doPnpmPublishVersions.ts
index c9e671aa8dbf588a9e7e5cbe1ef3c9024cbfa0dc..854753e23d8e1a39712b5bd2c3f2e7912e5a9bf2 100644
--- a/src/pnpm/doPnpmPublishVersions.ts
+++ b/src/pnpm/doPnpmPublishVersions.ts
@@ -25,9 +25,12 @@ export default async function (
     getOptionalFlagArgument("--filter", filters),
     ...getIncludeFilters(forcePackages),
     ...getExclusionFilters(excludePackages),
+    "exec",
+    "npm", // https://github.com/pnpm/pnpm/issues/11098
     "publish",
     getOptionalFlagArgument("--tag", tag),
-    getOptionalFlagArgument("--publish-branch", branch),
+    "--access", "public",
+    "--provenance"
   );
   await spawn("pnpm", pnpmArgs, pnpmLogger);
 }
