diff --git a/cli/index.cjs.map b/cli/index.cjs.map
deleted file mode 100644
index 69409e81e966203d37f1e0d3273c0e57ad1a02f2..0000000000000000000000000000000000000000
diff --git a/cli/index.js.map b/cli/index.js.map
deleted file mode 100644
index a2d522c60af5b8d83c8c82d9445be816e6ddbebe..0000000000000000000000000000000000000000
diff --git a/compiler/stencil.js.map b/compiler/stencil.js.map
deleted file mode 100644
index 6d73160eab5a683c3e55cf477fdd802772e8800b..0000000000000000000000000000000000000000
diff --git a/internal/index.d.ts b/internal/index.d.ts
index cc1ddef6a46e294b04fd3752e7e5139c84ccfb1d..be068a02be17a19b61b81bbca2f001a2cdd671d8 100644
--- a/internal/index.d.ts
+++ b/internal/index.d.ts
@@ -1,4 +1,3 @@
-/// <reference path="./stencil-ext-modules.d.ts" />
 export * from './stencil-private';
 export * from './stencil-public-compiler';
 export * from './stencil-public-runtime';
